tests/hypervisor/test_esx.py

```diff
@@ -5,40 +5,305 @@
 :caseautomation: Automated
 """
 import pytest
-from virtwho import logger
+from virtwho import REGISTER
+from virtwho import RHEL_COMPOSE
+from virtwho import HYPERVISOR
+from virtwho.configure import VirtwhoHypervisorConfig
 
 
+@pytest.mark.usefixtures('function_virtwho_d_conf_clean')
+@pytest.mark.usefixtures('debug_true')
+@pytest.mark.usefixtures('globalconf_clean')
 class TestEsx:
-    @pytest.mark.tier1
-    def test_hostname_option(self):
-        """Just a demo
+    @pytest.mark.tier2
+    def test_type(self, virtwho, function_hypervisor, esx_assertion):
+        """Test the type= option in /etc/virt-who.d/test_esx.conf
 
-        :title: virt-who: esx: test hostname option
-        :id: fb1f5dec-89c7-41e7-a15b-52b843f6f590
+        :title: virt-who: esx: test type option
+        :id: 0f2fcf4f-ac60-4c3c-905a-fabe22536ab2
         :caseimportance: High
-        :tags: tier1
+        :tags: tier2
         :customerscenario: false
         :upstream: no
         :steps:
-            1.
+            1. Configure type option with invalid value.
+            2. Disable the type option in the config file.
+            3. Create another valid config file
+            4. Update the type option with null value
+
         :expectedresults:
-            1.
+            1. Failed to run the virt-who service, find error messages
+            2. Find error message: Error in libvirt backend
+            3. Find error message, the good config works fine
+            4. The good config works fine
         """
-        logger.info("Succeeded to run the 'test_hostname_option'")
+        # type option is invalid value
+        validate_values = esx_assertion['type']
+        validate_values_list = list(validate_values['invalid'].keys())
+        for value in validate_values_list:
+            function_hypervisor.update('type', value)
+            result = virtwho.run_service()
+            assert (result['error'] is not 0
+                    and result['send'] == 0
+                    and result['thread'] == 0)
+            if 'RHEL-9' in RHEL_COMPOSE:
+                assert validate_values['invalid'][f'{value}'] in result['error_msg']
+            else:
+                assert validate_values['non_rhel9'] in result['error_msg']
+
+        # type option is disable
+        function_hypervisor.delete('type')
+        result = virtwho.run_service()
+        assert (result['error'] is not 0
+                and result['send'] == 0
+                and result['thread'] == 1
+                and validate_values['disable'] in result['error_msg'])
+
+        # type option is disable but another config is ok
+        new_file = '/etc/virt-who.d/new_config.conf'
+        section_name = 'virtwho-config'
+        new_hypervisor = VirtwhoHypervisorConfig(HYPERVISOR, REGISTER, new_file, section_name)
+        new_hypervisor.create()
```

Comment:
Could we perfect to use the `function_hypervisor.create(config_name=new_file, section=section_name)`.

Reply:
No, we shouldn't.  As they are 2 different instances and note that we still need the function_hypervirsor to update the option such as `function_hypervisor.update('type', '')`, if we would like to update another, should use `new_hypervisor.update('type', 'esx')`

Reply:
yes, make sense for me and thanks for your explanation, but I have another suggestion, to make a fixture for the `new_hypervisor` because several cases in each hypervisor will use it.